LUA: Add a function to retrieve the landed state

muramura opened the pull request:
I want to get the value of LANDED_STATE to determine the landing status in LUA. I will add a method to retrieve the LANDED_STATE. This ticket is related to #27221 .

RESULT
![Screenshot from 2024-06-02 15-32-46](https://github.com/ArduPilot/ardupilot/assets/646194/6bf31132-69ce-4223-b799-d9d3aec5704e)

LUA SCRIPT
![Screenshot from 2024-06-02 17-21-16](https://github.com/ArduPilot/ardupilot/assets/646194/ea20d4d6-15a4-4e55-b9b5-8f39066b79d7)

IamPete1 commented on libraries/AP_Scripting/docs/docs.lua:
```diff
@@ -2387,6 +2387,10 @@ function vehicle:is_taking_off() end
 ---@return boolean
 function vehicle:is_landing() end
 
+-- desc
+---@return integer
+function vehicle:get_landed_state() end
```
Full descriptions please. 

You can enumerate the return states like this:

https://github.com/ArduPilot/ardupilot/blob/eb8eeba8516f5fc3a83b82a7f03235c2ca6578ae/libraries/AP_Scripting/docs/docs.lua#L1416-L1423

olliw42 commented:
this is a bit strange. There is already a landed_state() in GCS_MAVlink which does exactly the same, so it's a good part code duplication. Also, why should only Copter get this lua. 

Note that there had been this PR https://github.com/ArduPilot/ardupilot/pull/25459, proposing to move em to the vehicle class, which should also adress this here.

olliw42 commented:
you do not need to use either to call it in lua
the function you define newly here is exactly identical to the function I mentioned is already there. Whatever name you give it, it doesn't change that fact :)

muramura commented:
I have implemented the process to determine the LAND state in the COPTER.
By doing this, multiple methods are no longer needed.
I have added a description of the return value to the LUA function.
